index.js

```diff
@@ -8587,37 +8607,54 @@ async function mergeBlobs({
   const type = 'blob';
   // Compute the new mode.
   // Since there are ONLY two valid blob modes ('100755' and '100644') it boils down to this
-  const mode =
-    (await base.mode()) === (await ours.mode())
-      ? await theirs.mode()
-      : await ours.mode();
-  // The trivial case: nothing to merge except maybe mode
-  if ((await ours.oid()) === (await theirs.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
-  }
-  // if only one side made oid changes, return that side's oid
-  if ((await ours.oid()) === (await base.oid())) {
-    return { mode, path, oid: await theirs.oid(), type }
-  }
-  if ((await theirs.oid()) === (await base.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
+  let ourMode = await ours.mode();
+  debugger;
+  if (base !== null) {
+    const mode =
+      (await base.mode()) === (ourMode)
+        ? await theirs.mode()
+        : ourMode;
+    // The trivial case: nothing to merge except maybe mode
+    if ((await ours.oid()) === (await theirs.oid())) {
+      return { mode, path, oid: await ours.oid(), type }
+    }
+    // if only one side made oid changes, return that side's oid
+    if ((await ours.oid()) === (await base.oid())) {
+      return { mode, path, oid: await theirs.oid(), type }
+    }
+    if ((await theirs.oid()) === (await base.oid())) {
+      return { mode, path, oid: await ours.oid(), type }
+    }
```

Comment:
This chain of statements could result in the same async functions being called several times. Even if it's not a problem, it seems unnecessary even if the functions are very cheap to perform. I feel like it makes the logic more difficult to read as well as opposed to something like:

```typescript
    const [oursOid, theirsOid, baseOid] = [await ours.oid(), await theirs.oid(), await base.oid()];

    if (oursOid === theirsOid) {
      return { mode, path, oid: oursOid, type }
    }
    // if only one side made oid changes, return that side's oid
    if (oursOid === baseOid) {
      return { mode, path, oid: theirsOid, type }
    }
    if (theirsOid === baseOid) {
      return { mode, path, oid: oursOid, type }
    }
```

This same logic occurs a total of four times (I think). Maybe possible to break it out into a function of three parameters? If not, that's fine. I don't know the structure of this repo so maybe there's no common place for the function to go.

The first and last if-statements return the same object. Is it possible to combine them into one if-statement?

Reply:
I don't disagree, but we're planning to readopt the original npm package in the next few months, so all these changes are going to be practically abandoned, so in that case is it worth it to make quality of life improvements to this code?

Reply:
The best laid plans of mice and men often go awry.

Are we planning on upstreaming these changes and the other changes we've made? If not, how can we go back to the upstream repo since we rely on our modifications? I guess I'm not familiar with the modifications we've actually made.

Reply:
I read the PR description, but I don't think I actually understand the purpose of the PR, or what the description is actually saying these changes are for. 😕

Reply:
@aaron-coalesce updated the description, happy to discuss further/demo if that helps.

We are planning to convert back to the original repo to make use of their version of conflict handling as our current implementation of merge conflicts handling has left users generally confused.  So we're going to abandon our current version and rework a better UX